### DISCUSSION TITLE
Tuyalocal 2024.3.1 seemingly breaking HASS

### DISCUSSION THREAD

**Discussion by adamzeis** [make-all/tuya-local#1732] [General]
Not sure where else to ask, but I recently updated to 2024.3.1 and it seems to be breaking my installation. Everything is running fine, I've restarted multiple times to be sure, but once I update to 2024.3.1 then I'm not able to full access HASS. I can get to a main dashboard, but everything is extremely slow to respond. I can't get far enough in to initiate a restore, however, and I've had to reset my HA Yellow and do a full restore from scratch. I'm not sure how to access any logs that would tell me any underlying issues, but curious if anyone has experienced anything similar or can point me in the right direction for diagnosing. Much appreciated.

**Comment by make-all** [comment]
2024.2.1 to 2024.3.0 did involve a config entry migration, so if you have many tuya-local devices and are running off an SD card, you may be seeing performance problems due to a slow file system until the migration completes.

**Reply by adamzeis** [reply]
I have about a dozen Tuya devices and I'm running HA Yellow. Do you suggest I update and then just need to wait out the migration? Just want to make sure I understand correctly. Thanks.

**Reply by make-all** [reply]
I'm not sure what exactly is happening, but if tuya-local is indeed the cause, I would speculate that it is most likely the migration, which should be cleared in a few minutes.

**Reply by adamzeis** [reply]
Okay thanks. Good to know there's more happening than normal at least. I'll give it another go and let it sit a while and see how it goes. Appreciate it!

**Reply by adamzeis** [reply]
Made it far enough to get into the logs (though hanging quite a bit still so far) and seeing this error. Happy to file a bug report if that's best.

Logger: homeassistant.util.async_
Source: util/async_.py:171
First occurred: 9:56:18 AM (8 occurrences)
Last logged: 9:58:08 AM

Detected blocking call to sleep inside the event loop by custom integration 'tuya_local' at custom_components/tuya_local/device.py, line 77: self._api = tinytuya.Device(dev_id, address, local_key), please create a bug report at https://github.com/make-all/tuya-local/issues

**Reply by adamzeis** [reply]
I tried updating again earlier. Let everything sit afterward for about 4 hours with no improvement. Ending up doing a full restore again unfortunately. Not sure what else to do at this point aside from just not updating the integration.

**Reply by make-all** [reply]
The error message can be avoided by setting the IP address of your devices in the config.  It is possible that the startup optimisations in HA 2024.3 turned this from a harmless warning into a critical problem, but there are currently no other reports of such problems.

I personally have found HA 2024.3 performance to have gotten a lot worse with ZHA and broadlink devices, but my tuya-local devices are as responsive as ever.

**Reply by adamzeis** [reply]
Thanks for this. I updated my Tuya devices with IP addresses, and also disabled the few that aren't currently connected. Updated and everything seems to be working now as intended. Very much appreciate the help and insight in getting this sorted.
